controller/appcontroller.go

```diff
@@ -477,11 +477,22 @@ func (ctrl *ApplicationController) processAppOperationQueueItem() (processNext b
 	return
 }
 
+// shouldbeDeleted returns whether a given resource obj should be deleted on cascade delete of application app
 func shouldBeDeleted(app *appv1.Application, obj *unstructured.Unstructured) bool {
 	return !kube.IsCRD(obj) && !isSelfReferencedApp(app, kube.GetObjectRef(obj))
 }
 
-func (ctrl *ApplicationController) finalizeApplicationDeletion(app *appv1.Application) error {
+// Checks whether an object is permitted to be deleted according to the project restrictions
+func (ctrl *ApplicationController) isPermittedForDelete(app *appv1.Application, obj *unstructured.Unstructured) (bool, error) {
+	if proj, err := ctrl.getAppProj(app); err != nil {
```

Comment:
What about combining this into `shouldBeDeleted` it seems to me that your interpretation would be consistent?

Reply:
Hm, I had it initially together in `shouldBeDeleted()`, but decided to decouple them. Mainly for reasons of differentiating later on, i.e. if we want to cancel deletion if there are resources who are not permitted. So I'd really like to keep it the way it currently is.

Reply:
I thought you wanted to semantics to be the same?

Reply:
Hm, semantics of `shouldBeDeleted()` are untouched?

Reply:
I think the semantics of a theoretical func`shouldNotBeDeleted` would include the permission check, so conversely I feel like `shouldBeDeleted`. Does that make sense?